src/lib.rs

```diff
@@ -1418,18 +1433,25 @@ impl<'a> MatchTreeLevel<'a> {
         })
     }
 
+    /// Attempts to accumulate a `MatchTree` recursively up to an overall depth of `depth` layers,
+    /// with the immediate layer constructed based on a bundle of indexed choice-points ([`LevelBranch`]).
+    ///
+    /// If the depth limit has been reached without a decisive choice of which index to accept, returns None.
+    ///
+    /// Otherwise, returns a `Some`-wrapped `MatchTree` that is guaranteed to decide on a unique branch for
```

Comment:
Minor point but "returns a MatchTree" is probably sufficient, with "Some-wrapped" implied.

Reply:
I see your point. Will simplify this.